synapse/cores/common.py

```diff
@@ -719,6 +719,21 @@ def getJoinByProp(self, prop, valu=None, mintime=None, maxtime=None, limit=None)
         '''
         return tuple(self._getJoinByProp(prop, valu=valu, mintime=mintime, maxtime=maxtime, limit=limit))
 
+    def getJoinInProp(self, prop, values, mintime=None, maxtime=None, limit=None):
```

Comment:
Loving the idea ( the optimization it allows especially ) not a huge fan of the API name ( or setting the precedent of an API update per optimization ).  Lets make it a "by" handler named "in".  Thoughts?